nms/packages/magmalte/app/views/equipment/EnodebDetailConfig.js

```diff
@@ -34,9 +34,826 @@ import {colors, typography} from '../../theme/default';
 import {makeStyles} from '@material-ui/styles';
 import {useContext, useState} from 'react';
 import {useEnqueueSnackbar} from '@fbcnms/ui/hooks/useSnackbar';
+import {useReducer} from 'react';
 import {useRouter} from '@fbcnms/ui/hooks';
 
+import AddIcon from '@material-ui/icons/Add';
+import DeleteIcon from '@material-ui/icons/Delete';
+import EditIcon from '@material-ui/icons/Edit';
+import IconButton from '@material-ui/core/IconButton';
+
+import ActionTable from '../../components/ActionTable';
+import Box from '@material-ui/core/Box';
+import Dialog from '@material-ui/core/Dialog';
+import DialogActions from '@material-ui/core/DialogActions';
+import DialogContent from '@material-ui/core/DialogContent';
+import DialogTitle from '@material-ui/core/DialogTitle';
+import FormControl from '@material-ui/core/FormControl';
+import FormControlLabel from '@material-ui/core/FormControlLabel';
+import InputLabel from '@material-ui/core/InputLabel';
+import MenuItem from '@material-ui/core/MenuItem';
+import Select from '@material-ui/core/Select';
+import Switch from '@material-ui/core/Switch';
+import TextField from '@material-ui/core/TextField';
+
+function reducer(state, action) {
+  const index = state.indexOf(action.item);
+
+  switch (action.type) {
+    case 'add':
+      const arr = [...state, action.item];
+      if (action.fn) action.fn(arr);
+      return arr;
+    case 'modify':
+      Object.assign(state[index], action.new);
+      if (action.fn) action.fn(state);
+      return [...state];
+    case 'remove':
+      const arrD = [...state.slice(0, index), ...state.slice(index + 1)];
+      if (action.fn) action.fn(arrD);
+      return arrD;
+    default:
+      throw new Error();
+  }
+}
+
+function cellReducer(state, action) {
+  const index = state.indexOf(action.item);
+
+  switch (action.type) {
+    case 'add':
+      const arr = [...state, action.item];
+      if (action.fn) action.fn(arr);
+      return arr;
+    case 'modify':
+      Object.assign(state[index], action.new);
+      if (action.fn) action.fn(state);
+      return [...state];
+    case 'remove':
+      const arrD = [...state.slice(0, index), ...state.slice(index + 1)];
+      if (action.fn) action.fn(arrD);
+      return arrD;
+    default:
+      throw new Error();
+  }
+}
+
+function FreqFormDialog(props) {
```

Comment:
React components should be documented with comments explaining what the component is for. Also make sure to include Flow typing annotations for argument types, and return types (if applicable).